examples/bri-3/src/bri/workgroup/workgroups/agents/workgroupStorage.agent.ts

```diff
@@ -0,0 +1,220 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { BpiSubject } from '../../../identity/bpiSubjects/models/bpiSubject';
+import { PrismaService } from '../../../../../prisma/prisma.service';
+import { Workflow } from '../../workflows/models/workflow';
+import { Workstep } from '../../worksteps/models/workstep';
+import { Workgroup } from '../models/workgroup';
+import { WORKGROUP_NOT_FOUND_ERR_MESSAGE } from '../api/err.messages';
+
+@Injectable()
+export class WorkgroupStorageAgent extends PrismaService {
+  async getWorkgroupById(id: string): Promise<Workgroup> {
+    const workgroupModel = await this.workgroup.findUnique({
+      where: { id: id },
+      include: {
+        worksteps: true,
+        administrators: true,
+        participants: true,
+        workflows: {
+          include: { worksteps: true },
+        },
+      },
+    });
+
+    if (!workgroupModel) {
+      throw new NotFoundException(WORKGROUP_NOT_FOUND_ERR_MESSAGE);
+    }
+
+    return new Workgroup(
+      workgroupModel.id,
+      workgroupModel.name,
+      workgroupModel.administrators.map((w) => {
```

Comment:
w is a confusing name for the local var since we are dealing with bpiSubjects here. a or bp?

Reply:
Fixed